### DISCUSSION TITLE
Mapping generics

### DISCUSSION THREAD

**Discussion by PieterBoeren** [nartc/mapper#357] [Q&A] [answered]
Maybe I just don't get the syntax (hopefully ;)), but currently I have to do something like:

`mapper.mapArray(dtoArray, SomeViewModel, SomeDto)`

But is it possible to do something like (inside a class where the generic types are defined, like "class X<TViewModel, TDto>"):

`mapper.mapArray<TViewModel, TDto>(dtoArray)`

(edit: I'm using profiles to define the mappings)

**Comment by micalevisk** [comment, marked as answer]
you can't use it like that because `<TViewModel, TDto>` is just typing stuff that cannot be read at compile nor at run time (generally speaking), thus you need to tell to AutoMapper what are the source and destination classes

See https://github.com/microsoft/TypeScript/wiki/FAQ#what-is-type-erasure

**Reply by PieterBoeren** [reply]
Clear! Thanks for the very quick answer!

**Reply by nartc** [reply]
Hi @PieterBoeren, what @micalevisk said is partly correct. The type can be read at compilation time but then we'd want the consumers to use this library (`AutoMapper`) in a manner that is as natural to any TypeScript project as possible. Having code that depends on Compilation Process is always tricky to setup for consumers :)
